Nos/Views/AuthorListView.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  AuthorListView.swift
+//  Nos
+//
+//  Created by Martin Dutra on 29/3/23.
+//
+
+import Foundation
+import SwiftUI
+
+struct AuthorListView: View {
+    @Environment(\.managedObjectContext) private var viewContext
+
+    @State private var authors: [Author]?
+
+    @State private var filteredAuthors: [Author]?
+
+    @StateObject private var searchTextObserver = SearchTextFieldObserver()
+
+    var didSelectGesture: ((Author) -> Void)?
+
+    var body: some View {
+        ScrollView(.vertical) {
+            LazyVStack {
+                if let authors = filteredAuthors {
+                    ForEach(authors) { author in
+                        AuthorCard(author: author) {
+                            didSelectGesture?(author)
+                        }
+                        .padding(.horizontal)
+                    }
+                } else {
+                    ProgressView()
+                }
+            }
+            .padding(.top)
+        }
+        .background(Color.appBg)
+        .nosNavigationBar(title: .authors)
+        .onChange(of: searchTextObserver.debouncedText) { newValue in
+            search(for: newValue)
+        }
+        .searchable(text: $searchTextObserver.text, placement: .navigationBarDrawer(displayMode: (.always)))
+        .disableAutocorrection(true)
+        .task {
+            refreshAuthors()
+        }
+    }
+
+    private func refreshAuthors() {
+        let request = Author.allAuthorsRequest(muted: false)
```

Comment:
Let's filter out authors without a name or display name here. When I open this view it shows 90% npubs with no names which isn't very helpful.

Reply:
👌

Reply:
👌